invoicer_project/static/js/add_client_request.js

```diff
@@ -0,0 +1,274 @@
+    for (let item of returnAllFields()) {
+        item.removeAttribute("error");
+        item.removeAttribute("errorText");
+    }
+}
+
+function allAreFalse(object) {
+    for (let key in object) {
+        if (Boolean(object[key]) === true) {
+            return false;
+        }
+    }
+    return true;
+}
+
+function validateClientAdd() {
+    removeAllErrorAttributes();
+    setMaxFieldContainerHeights();
+   return {
+        'nameValidationResult': validateNameAndSurnameAsStrings(nameField.value),
+        'surnameValidationResult': validateNameAndSurnameAsStrings(surnameField.value),
+        'emailValidationResult': validateEmail(emailField.value),
+        'telephoneValidationResult': validateTelephone(telephoneField.value),
+        'zipValidationResult': validateZip(zipField.value),
+        'countryValidationResult': validateCountry(countryField.value),
+        'cityValidationResult': validateCity(cityField.value),
+        'addressValidationResult': validateAddress(addressField.value),
+    };
+}
+
+function validateNameAndSurnameAsStrings(strToValidate) {
+    let strValidationResult
+    if (!strToValidate) {
+        strValidationResult = "This field can't be empty";
+    } else if (strToValidate.includes(' ')) {
+        strValidationResult = "No whitespaces";
+    } else if (strToValidate.length > nameSurnMaxLength) {
+        strValidationResult = `Max length – ${nameSurnMaxLength} chars`;
+    } else if (!(/^[a-z]+$/i.test(strToValidate))) {
+        strValidationResult = "Only latin letters";
+    } else if (!(/[A-Z]/.test(strToValidate.charAt(0)))) {
+        strValidationResult = "Has to begin with capital";
+    } else if (strToValidate.replace(/[^A-Z]/g, "").length > 1) {
+        strValidationResult = "No more than one capital"
+    } else if (!/^[A-Z][a-z]+$/.test(strToValidate)) {
+        strValidationResult = "At least one lowercase";
+    } else {
+        strValidationResult = '';
+    }
+    return strValidationResult;
+}
+
+function validateEmail(emailToValidate) {
+    let isEmailValid
+    if (emailToValidate === '') {
+        isEmailValid = "This field can't be empty";
+    } else if (emailToValidate.includes(' ')) {
+        isEmailValid = "No whitespaces";
+    } else if (!(/^[a-zA-Z0-9.]{3,20}@(?:[a-zA-Z0-9]{2,20}\.){1,30}[a-zA-Z]{2,10}$/.test(emailToValidate))) {
+        isEmailValid = "Invalid email format";
+    } else {
+        isEmailValid = '';
+    }
+    return isEmailValid;
+}
+
+function validateTelephone(telephoneToValidate) {
+    let isTelephoneValid
+    if (telephoneToValidate === '') {
+        isTelephoneValid = "This field can't be empty";
+    } else if (telephoneToValidate.includes(' ')) {
+        isTelephoneValid = "No whitespaces";
+    } else if (!(/^\+?1?\d{9,15}$/.test(telephoneToValidate))) {
+        isTelephoneValid = "Invalid telephone format";
+    }  else {
+        isTelephoneValid = '';
+    }
+    return isTelephoneValid;
+}
+
+function validateZip(zipToValidate) {
```

Comment:
Looks like methods `validateTelephone`, `validateEmail` and `validateZip` have very similar code except pattern. I guess it  is possible to make one function insteand

Reply:
Done